### DISCUSSION TITLE
Crazyflie has to be restarted after motion_commander_demo script runs

### DISCUSSION THREAD

**Discussion by dbaldwin** [bitcraze/discussions#969] [Q&A] [answered]
Hi,

I'm working my way through CRTP by using the motion_commander_demo.py example as a reference. I've modified the script to do a basic takeoff and land like this in this code snippet:

```
mc = MotionCommander(scf)
mc.take_off()
time.sleep(5)
mc.land()
```

It works well but for some reason if I run it again Crazyflie will ignore any packets. I have to restart my drone to get the script to work again. I've experienced this behavior with two different Crazyflies. Is there something I should be checking/resetting to get it to run subsequent times?

Thanks.

**Comment by knmcguire** [comment]
I can confirm. With the old way:
```
    with SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf:
        with MotionCommander(scf) as mc:
            #mc.take_off()
            time.sleep(5)
            #mc.land()
```
The mc.take_off and land are commented out as they already are initiated with the initialization and closing of the motioncommander instance. Then I don't see this problem when reconnecting

With the similar script as yours:

```
    with SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf:
      mc = MotionCommander(scf)
      mc.take_off()
      time.sleep(5)
      mc.land()
```

Upon reconnection it does indeed not respond. Eventhough the [__enter__ and __exit__ functions](https://github.com/bitcraze/crazyflie-lib-python/blob/8cd8523a84d1dc9188c575ccce0f30b93cf68c9d/cflib/positioning/motion_commander.py#L122-L127) do exactly the same... maybe something to do with the threads not being properly stopped? 

This is more of an issue actually so I'll create one!

**Reply by knmcguire** [reply]
The issue is here https://github.com/bitcraze/crazyflie-lib-python/issues/419. It's a bit of figuring out what the difference between the with statement (context manager) or initializing the object like you did with the take off and land. But we should probably continue the investigation over there.

**Comment by krichardsson** [comment, marked as answer]
I don't think there is any difference in the two flavors of calling the Motion commander, if you see a difference it is perhaps due to timing?

Anyway, fixed in https://github.com/bitcraze/crazyflie-lib-python/pull/420

**Comment by dbaldwin** [comment]
Thank you for the quick response and resolution. This is working well for me. It's refreshing to work with a company that is responsive and provides incredible support for their product!

**Reply by knmcguire** [reply]
you're welcome :) Support is one of our main priorities. Especially since this is a development product that is what we think is important. Some might argue that we maybe respond a bit too quick :P 

Glad that it solved it for you! We will be working on a release this month so soon it will be part of the pip packages as well. 

I've also converted your discussion to a Q&A, such that you can select an answer instead of closing them since people will then still see the answer. I've already selected kristoffers' for you.
